onnxruntime/hosting/server_configuration.h

```diff
@@ -0,0 +1,71 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#ifndef ONNXRUNTIME_HOSTING_SERVER_CONFIGURATION_H
+#define ONNXRUNTIME_HOSTING_SERVER_CONFIGURATION_H
+
+#include <thread>
+#include <fstream>
+
+#include "boost/program_options.hpp"
+
+namespace onnxruntime {
+namespace hosting {
+
+namespace po = boost::program_options;
+
+// Wrapper around Boost program_options and should provide all the functionality for options parsing
+// Provides sane default values
+class ServerConfiguration {
+ public:
+  ServerConfiguration() {
+    desc.add_options()("help,h", "Shows a help message and exits");
+    desc.add_options()("model_path,m", po::value(&model_path)->required(), "Path to ONNX model");
+    desc.add_options()("address,a", po::value(&address)->default_value(address), "The base HTTP address");
+    desc.add_options()("port,p", po::value(&port)->default_value(port), "HTTP port to listen to requests");
+    desc.add_options()("threads,t", po::value(&threads)->default_value(threads), "Number of http threads");
+  }
+
+  void ParseInput(int ac, char** av) {
+    try {
+      po::store(po::command_line_parser(ac, av).options(desc).run(), vm);  // can throw
+
+      if (ContainsHelp()) {
+        PrintHelp(std::cout, full_desc);
+        return exit(EXIT_SUCCESS);
+      }
+
+      po::notify(vm);  // throws on error, so do after help
+    } catch (const po::error& e) {
+      PrintHelp(std::cerr, e.what());
+      return exit(EXIT_FAILURE);
+    } catch (const std::exception& e) {
+      PrintHelp(std::cerr, e.what());
+      return exit(EXIT_FAILURE);
+    }
+  }
+
+  bool ContainsHelp() const {
+    return vm.count("help") || vm.count("h");
```

Comment:
Is it case sensitive or insensitive? We may need to explicitly call it out in the help message.

Reply:
It's case sensitive like all Unix utilties. Even if they get the case wrong they'll get:

```
unrecognised option '--Help'
Allowed options:
  -h [ --help ]                   Shows a help message and exits
  -m [ --model_path ] arg         Path to ONNX model
  -a [ --address ] arg (=0.0.0.0) The base HTTP address
  --http_port arg (=5000)         HTTP port to listen to requests
  --num_http_threads arg (=8)     Number of http threads

```